lib/upipe-ts/upipe_ts_scte_common.c

```diff
@@ -0,0 +1,299 @@
+                }
+
+                uint8_t upid_type = scte35_seg_desc_get_upid_type(desc);
+                uint8_t upid_length = scte35_seg_desc_get_upid_length(desc);
+                const uint8_t *upid = scte35_seg_desc_get_upid(desc);
+                uint8_t type_id = scte35_seg_desc_get_type_id(desc);
+                uint8_t num = scte35_seg_desc_get_num(desc);
+                uint8_t expected = scte35_seg_desc_get_expected(desc);
+                if (upid_type || upid_length) {
+                    uref_ts_scte35_desc_seg_set_upid_type(out, upid_type);
+                    uref_ts_scte35_desc_seg_set_upid_type_name(
+                        out, scte35_seg_desc_upid_type_to_str(upid_type));
+                    uref_ts_scte35_desc_seg_set_upid_length(
+                        out, upid_length);
+                    uref_ts_scte35_desc_seg_set_upid(
+                        out, upid, upid_length);
+                }
+                uref_ts_scte35_desc_seg_set_type_id(out, type_id);
+                uref_ts_scte35_desc_seg_set_type_id_name(
+                    out, scte35_seg_desc_type_id_to_str(type_id));
+                uref_ts_scte35_desc_seg_set_num(out, num);
+                uref_ts_scte35_desc_seg_set_expected(out, expected);
+                if (scte35_seg_desc_has_sub_num(desc)) {
+                    uint8_t sub_num =
+                        scte35_seg_desc_get_sub_num(desc);
+                    uref_ts_scte35_desc_seg_set_sub_num(out, sub_num);
+                }
+                if (scte35_seg_desc_has_sub_expected(desc)) {
+                    uint8_t sub_expected =
+                        scte35_seg_desc_get_sub_expected(desc);
+                    uref_ts_scte35_desc_seg_set_sub_expected(
+                        out, sub_expected);
+                }
+            }
+            break;
+        }
+    }
+    return out;
+}
+
+/** @This export an uref describing a SCTE35 descriptor.
+ *
+ * @param upipe description structure of the caller
+ * @param uref uref to export
+ * @param desc pointer to the SCTE35 descriptor destination
+ * @return an error code
+ */
+int upipe_ts_scte_export_desc(struct upipe *upipe,
+                              struct uref *uref,
+                              uint8_t *desc)
+{
+    uint8_t tag;
+    UBASE_RETURN(uref_ts_scte35_desc_get_tag(uref, &tag));
+    uint64_t identifier;
+    UBASE_RETURN(uref_ts_scte35_desc_get_identifier(uref, &identifier));
+
+    uint32_t length = SCTE35_SPLICE_DESC_HEADER_SIZE;
+    switch (tag) {
+        case SCTE35_SPLICE_DESC_TAG_SEG: {
+            length += SCTE35_SEG_DESC_HEADER_SIZE;
+            uint64_t event_id;
+            UBASE_RETURN(uref_ts_scte35_desc_seg_get_event_id(uref, &event_id));
+            bool cancel = ubase_check(uref_ts_scte35_desc_seg_get_cancel(uref));
+            bool has_delivery_not_restricted =
+                scte35_seg_desc_has_delivery_not_restricted(desc);
```

Comment:
Shouldn't it be ubase_check(uref_ts_scte35_desc_has_delivery_not_restricted(uref) ?

Reply:
Yes, fixed